integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/altinn/SvarInnPutMessageForwarder.java

```diff
@@ -0,0 +1,147 @@
+
+import lombok.RequiredArgsConstructor;
+import net.logstash.logback.marker.Markers;
+import no.difi.meldingsutveksling.MessageInformable;
+import no.difi.meldingsutveksling.ServiceIdentifier;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.ks.svarinn.Forsendelse;
+import no.difi.meldingsutveksling.ks.svarinn.SvarInnFieldValidator;
+import no.difi.meldingsutveksling.ks.svarinn.SvarInnPutMessageBuilder;
+import no.difi.meldingsutveksling.ks.svarinn.SvarInnService;
+import no.difi.meldingsutveksling.logging.Audit;
+import no.difi.meldingsutveksling.nextmove.ConversationDirection;
+import no.difi.meldingsutveksling.noarkexchange.NoarkClient;
+import no.difi.meldingsutveksling.noarkexchange.PutMessageRequestFactory;
+import no.difi.meldingsutveksling.noarkexchange.logging.PutMessageResponseMarkers;
+import no.difi.meldingsutveksling.noarkexchange.schema.PutMessageRequestType;
+import no.difi.meldingsutveksling.noarkexchange.schema.PutMessageResponseType;
+import no.difi.meldingsutveksling.noarkexchange.schema.core.DokumentType;
+import no.difi.meldingsutveksling.receipt.Conversation;
+import no.difi.meldingsutveksling.receipt.ConversationService;
+import no.difi.meldingsutveksling.receipt.MessageStatusFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.time.Clock;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static no.difi.meldingsutveksling.receipt.ReceiptStatus.INNKOMMENDE_LEVERT;
+import static no.difi.meldingsutveksling.receipt.ReceiptStatus.INNKOMMENDE_MOTTATT;
+
+@RequiredArgsConstructor
+@ConditionalOnProperty(name = "difi.move.feature.enableDPF", havingValue = "true")
+@Component
+public class SvarInnPutMessageForwarder implements Consumer<Forsendelse> {
+
+    private final IntegrasjonspunktProperties properties;
+    private final ConversationService conversationService;
+    private final SvarInnService svarInnService;
+    private final NoarkClient localNoark;
+    private final NoarkClient fiksMailClient;
+    private final MessageStatusFactory messageStatusFactory;
+    private final PutMessageRequestFactory putMessageRequestFactory;
+    private final Clock clock;
+
+    @Override
+    public void accept(Forsendelse forsendelse) {
+        SvarInnPutMessageBuilder builder = new SvarInnPutMessageBuilder(forsendelse, putMessageRequestFactory);
+        svarInnService.getAttachments(forsendelse).forEach(builder::streamedFile);
+        PutMessageRequestType putMessage = builder.build();
+
+        if (builder.getDokumentTypeList().isEmpty()) {
+            Audit.error("Zipfile is empty: skipping message", Markers.append("fiks-id", forsendelse.getId()));
+            return;
+        }
+
+        Conversation c = conversationService.registerConversation(new MessageInformable() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 30 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1188)